Compilation error are no longer caught by data processor

With 2.4.1 when defining an invalid finder like:

```java
@JdbcRepository(dialect = Dialect.H2)
interface FooRepository extends CrudRepository<Foo, Long> {
    Foo findByNom(String name);
}
```

Results in 

```
> Task :compileJava FAILED
java.lang.IllegalArgumentException: Cannot query entity [Foo] on non-existent property: nom
        at io.micronaut.data.processor.visitors.finders.DynamicFinder.verifyFinderParameter(DynamicFinder.java:346)
        at io.micronaut.data.processor.visitors.finders.DynamicFinder.buildMatchInfo(DynamicFinder.java:226)
        at io.micronaut.data.processor.visitors.RepositoryTypeElementVisitor.visitMethod(RepositoryTypeElementVisitor.java:246)
        at io.micronaut.annotation.processing.visitor.LoadedVisitor.visit(LoadedVisitor.java:147)
```

This exception should never propagate out of the compiler as it has already been reported as an error:

```
/Users/graemerocher/dev/micronaut/apps/data-app/src/main/java/data/app/FooRepository.java:11: error: Unable to implement Repository method: FooRepository.findByNom(String name). Cannot query entity [Foo] on non-existent property: nom
    Foo findByNom(String name);
        ^
Note: Creating bean classes for 2 type elements
1 error
```

--- a/data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java
+++ b/data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java
@@ -248,7 +248,6 @@ public class RepositoryTypeElementVisitor implements TypeElementVisitor<Reposito
                             this.failing = true;
                             return;
                         } catch (Exception e) {
-                            e.printStackTrace();
                             matchContext.fail(e.getMessage());
                             this.failing = true;
                             return;

